test.js

```diff
@@ -7,668 +7,465 @@ const Next = require('next')
 const pino = require('pino')
 const proxyquire = require('proxyquire')
 const sinon = require('sinon')
+const { Agent, setGlobalDispatcher, request } = require('undici')
 
-test('should construct next with proper environment', t => {
-  t.plan(2)
+const port = 0
+const agent = new Agent({
+  keepAliveTimeout: 1,
+  keepAliveMaxTimeout: 1
+})
+
+setGlobalDispatcher(agent)
 
+test('should construct next with proper environment', async t => {
   process.env.NODE_ENV = 'production'
 
-  let options
   const dev = process.env.NODE_ENV !== 'production'
+  const app = Next(Object.assign({}, { dev }))
+
+  await app.prepare()
 
   t.equal(dev, false)
+  t.equal(app.dev, undefined)
 
-  const app = Next(Object.assign({}, { dev }, options))
-  app.prepare()
-    .then(() => {
-      t.equal(app.dev, undefined)
-    })
   app.close()
 })
 
-test('should return an html document', t => {
-  t.plan(3)
```

Comment:
Because having translated the tests to async/await there aren't any children/nested tests, so there's no real need to specify the exact number of tests beforehand.

Reply:
I agree with this approach. t.plan is really only needed when making assertions in callbacks. it makes the test harder to maintain so I'd only use it in those scenarios and replace it with async/await otherwise